synapse/storage/databases/main/transactions.py

```diff
@@ -43,15 +43,36 @@
 SENTINEL = object()
 
 
-class TransactionStore(SQLBaseStore):
+class TransactionWorkerStore(SQLBaseStore):
+    def __init__(self, database: DatabasePool, db_conn, hs):
+        super().__init__(database, db_conn, hs)
+
+        self._clock.looping_call(self._start_cleanup_transactions, 30 * 60 * 1000)
+
+    def _start_cleanup_transactions(self):
```

Comment:
While we're here should we make things a bit more consistent by using `wrap_as_background_process`, rather than wrapping these things in background processes in a myriad of different ways?

Reply:
I had wondered that. I think it makes sense to do...as long as it is a separate commit. I'll go ahead and do that.

Reply:
I went ahead and did this, also included this for the changes from #8369 to be consistent.